lib/Runtime/Debug/TTEventLog.cpp

```diff
@@ -31,17 +30,18 @@ namespace TTD
             //Update the time elapsed since a snapshot if needed
             if(cfAction->CallbackDepth == 0)
             {
-                double elapsedTime = (cfAction->AdditionalInfo->EndTime - cfAction->AdditionalInfo->BeginTime);
+                double elapsedTime = (elog->GetCurrentWallTime() - this->m_beginTime);
```

Comment:
Is it worth adding an assert that this is greater than 0? Or perhaps clamping it?